libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -143,7 +143,19 @@ func (p *addHostMetadata) loadData() error {
 		return err
 	}
 
-	data := host.MapHostInfo(features.FQDN(), h.Info())
+	hostname := h.Info().Hostname
+	if features.FQDN() {
+		fqdn, err := h.FQDN()
+		if err != nil {
+			// FQDN lookup is "best effort".  We log the error, fallback to
+			// the OS-reported hostname, and move on.
+			p.logger.Infof("unable to lookup FQDN: %s", err.Error())
+		} else {
+			hostname = fqdn
+		}
+	}
+
+	data := host.MapHostInfo(h.Info(), hostname)
```

Comment:
Depends on https://github.com/elastic/elastic-agent-system-metrics/pull/81 being merged and dependency on `elastic-agent-system-metrics` being bumped up.